### ISSUE TITLE
Bring common controller config items out into interface

### ISSUE THREAD

**Issue by shreyas-s-rao** [gardener/etcd-druid#937]
**How to categorize this issue?**

/area quality
/kind enhancement

**What would you like to be added**:

Create interface for controller `Config`, which includes `Workers` and `FeatureGates`, and collect all `CaptureFeatureActivations` methods into a single method on the interface, which is called for all controllers by default.

**Why is this needed**:
Harmonize configurations for all the controllers, and the webhooks as well to some extent. For instance, since all controllers specify `Workers` in their configs, this can be brought out as a common configuration item. Similarly, `FeatureGates` is used by all controllers and webhooks, so that can be brought out as a common config item as well.

**Comment by unmarshall** [comment]
@shreyas-s-rao Will this not be solved by https://github.com/gardener/etcd-druid/issues/970?

**Comment by shreyas-s-rao** [comment]
I don't think so. #970 talks about configuring the overall druid operator, but this issue is more about the code-level, where we bring common elements of the controller config out, which each controller can then extend. This allows to reduce code duplication across controllers.

**Comment by unmarshall** [comment]
Actually when implementing #970 we will be introducing golang types for etcd-druid operator configuration which will include configuration for individual reconcilers as well. So i still see a large overlap.

**Comment by shreyas-s-rao** [comment]
Ok, makes sense. Then we take up the task of bringing out common fields for controllers in the other issue.
/close
